### PR TITLE
Decomposition of model-specific optimizer functions

### PR DESCRIPTION
**Pull request by sebastianober**
Following #272 it would be good to decompose model-specific optimizer functions into their own files corresponding to each model.

### CODE REVIEW

File: trieste/models/optimizer.py [hunk 1]
```diff
@@ -161,14 +160,6 @@ def _create_tf_optimizer(
...
-def _create_scipy_optimizer(
-    optimizer: gpflow.optimizers.Scipy,
-    optimizer_args: Dict[str, Any],
-) -> Optimizer:
-    return Optimizer(optimizer, **optimizer_args)
-
-
 @singledispatch
 def create_loss_function(
     model: gpflow.models.GPModel, dataset: TrainingData, compile: bool = False
```
Comment (hstojic) [inline trieste/models/optimizer.py]: additional gpflow leftovers...

Comment (sebastianober) [reply trieste/models/optimizer.py]: Should be fixed now

**Comment by sebastianober** [comment]
TODO: decide what to do about Keras vs TF optimizers - do we want a `KerasOptimizer` class? Perhaps this might be something that would be better for a new issue?

**Comment by sebastianober** [comment]
@hstojic Any ideas on what you want to do with this? Should we just leave it with my GPflux PR?

**Comment by hstojic** [comment]
> @hstojic Any ideas on what you want to do with this? Should we just leave it with my GPflux PR?

I'll pick this up and sort out the PR, this week or the next

**Comment by hstojic** [comment]
new PR #372 is replacing this one

**Pull request closed without merging**